indico/types/dataset.py

```diff
@@ -0,0 +1,28 @@
+from typing import List
+
+from indico.types.base import BaseType
+from indico.types.datafile import Datafile
+
+class DataColumn(BaseType):
+    id: int
+    name: str
+
+class LabelSet(BaseType):
+    id: int 
+    name: str
+
+class Dataset(BaseType):
+    id: int
+    name: str   
+    rowCount: int
+    status: str
+    permissions: str
+    files: List[Datafile]
+    labelsets: List[LabelSet]
+    datacolumns: List[DataColumn]
+
+    def labelset_by_name(self, name: str) -> LabelSet:
+        return next(l for l in self.labelsets if l.name == name)
```

Comment:
Why is next needed here? vs. using `filter` on the list for example or just using a list comprehension?